Silly Lava Wombat

Medium

# `SubProxy::exec()` does not handle return data

### Summary

There is a missing check on the [`out`](https://github.com/sherlock-audit/2024-06-makerdao-endgame/blob/dba30d7a676c20dfed3bda8c52fd6702e2e85bb1/endgame-toolkit/src/SubProxy.sol#L76) return data in the `exec` function 

### Root Cause

In [`SubProxy::exec()` ](https://github.com/sherlock-audit/2024-06-makerdao-endgame/blob/dba30d7a676c20dfed3bda8c52fd6702e2e85bb1/endgame-toolkit/src/SubProxy.sol#L74)there is no check that handles the data parameter after the delegatecall, it is only the bool param that is handled

### Internal pre-conditions

_No response_

### External pre-conditions

_No response_

### Attack Path

_No response_

### Impact

No handling of the data param can lead to complications in the code execution.

### PoC

```solidity
    function exec(address target, bytes calldata args) external payable auth returns (bytes memory out) {
        bool ok;
        (ok, out) = target.delegatecall(args);
        require(ok, "SubProxy/delegatecall-error");
    }
```

### Mitigation

implement handling on the return data parameter after the delegate call